similar/arch/ogl/gr.cpp

```diff
@@ -1049,30 +1049,38 @@ static int do_pal_step;
 
 void ogl_do_palfx(void)
 {
-	// scale negative effect by 2.5 to match D1/D2 on GL
-	alast_r = abs(last_r) * (last_r < 0 ? 2.5 : 1.0);
-	alast_g = abs(last_g) * (last_g < 0 ? 2.5 : 1.0);
-	alast_b = abs(last_b) * (last_b < 0 ? 2.5 : 1.0);
-
 	GLfloat color_array[] = { 
-		alast_r, alast_g, alast_b, 1.0,
-		alast_r, alast_g, alast_b, 1.0,
-		alast_r, alast_g, alast_b, 1.0,
-		alast_r, alast_g, alast_b, 1.0
+		last_r, last_g, last_b, 1.0,
+		last_r, last_g, last_b, 1.0,
+		last_r, last_g, last_b, 1.0,
+		last_r, last_g, last_b, 1.0
 	};
 
 	OGL_DISABLE(TEXTURE_2D);
 
 	glEnableClientState(GL_VERTEX_ARRAY);
 	glEnableClientState(GL_COLOR_ARRAY);
  
-	if (do_pal_step)
+	if (do_pal_step && ((last_r <= 0) & (last_g <= 0) & (last_b <= 0))) {
+		// scale negative effect by 2.5 to match D1/D2 on GL
+		alast_r = abs(last_r) * 2.5;
+		alast_g = abs(last_g) * 2.5;
+		alast_b = abs(last_b) * 2.5;
+
+		color_array[ 0] = color_array[ 4] = alast_r;
```

Comment:
This does the right thing, but looks odd at first.  I'd prefer to order the logic so that the array is initialized with the right values, rather than patching it later.